astropy/stats/funcs.py

```diff
@@ -42,6 +40,51 @@
 """
 
 
+def _expand_dims(data, axis):
```

Comment:
To ensure that we do not forget to remove it, I would do
```
from astropy.utils.compat import NUMPY_LT_1_18

if NUMPY_LT_1_18:
    def _expand_dims(...)
        ...
else:
    from numpy import expand_dims as _expand_dims
```
(note that `NUMPY_LT_1_18` already exists; obviously, since https://github.com/numpy/numpy/pull/14051 isn't merged yet, this will temporarily break the numpy-dev build... another ping there may be worth it).

p.s. Even if you don't go for this, maybe just add `# NUMPY_LT_1_18`, since that way we are sure to eventually remove it.

Reply:
Actually, possibly this is better done where you actually use it... See note there. That probably is better & clearer.

Reply:
I think I'm going to just add `# NUMPY_LT_1_18` before the function definition *and* where I call `_expand_dims`.  Does that sound OK?  I'm not sure when the PR will be merged (I have no idea what's taking so long), and the numpy dev CI will fail in the meantime.

Reply:
I added the comments for now.  Let me know if that's not sufficient.

Reply:
please do add `NUMPY_LT_1_18` where stuff is actually defined, so the a braindead cleanup like the one I did in #9292  will easily find it and can just remove without a deeper dive into the code. 
If that suggestion was because of the private nature of the function, maybe consider to place it direcly in `astropy/utils/compat/numpycompat.py`, that shouldn't be used by users.

Reply:
Yep, I added that comment everywhere things would need to be changed.